vendor/signals.py

```diff
@@ -63,4 +65,120 @@ def stripe_delete_customer_signal(sender, instance, **kwargs):
+
+    pk_clause = query_builder.make_clause_template(
+        field='metadata',
+        key='pk',
+        value=instance.pk,
+        operator=query_builder.EXACT_MATCH,
+        next_operator=query_builder.AND
+    )
+
+    metadata_clause = query_builder.make_clause_template(
+        field='metadata',
+        key='site',
+        value=instance.site.domain,
+        operator=query_builder.EXACT_MATCH
+    )
+
+    query = query_builder.build_search_query(processor.stripe.Product, [pk_clause, metadata_clause])
+    query_result = processor.stripe_query_object(processor.stripe.Product, query)
+
+    if not processor.transaction_succeded:
+        logger.error(f"stripe_create_offer_signal: {processor.transactions_info}")
+        return None
+
+    if query_result:
+        coupons = processor.get_coupons()
+        coupon_matches = processor.match_coupons(coupons, instance)
+        product_id = query_result.data[0].id
+        price_id = processor.get_price_id_with_product(product_id)
+        instance.meta['stripe'] = {
+            'product_id': product_id,
+            'price_id': price_id
+        }
+
+        if coupon_matches:
+            instance.meta['coupon_id'] = coupon_matches[0].id
+
+        instance.save()
+
+        if len(coupon_matches) > 1:
+            logger.warning(
+                f"stripe_create_offer_signal: more than one coupon found for offer: {instance.pk} on site: {instance.site.domain}")
+
+        if len(query_result.data) > 1:
+            logger.warning(
+                f"stripe_create_offer_signal: more than one offer/product found for pk: {instance.pk} on site: {instance.site.domain}")
+        return None
+
+    product_data = processor.build_product(instance)
+    stripe_product = processor.stripe_create_object(processor.stripe.Product, product_data)
+    instance.meta['stripe'] = {'product_id': stripe_product.id}
+
+    price = instance.get_current_price_instance() if instance.get_current_price_instance() else None
+    msrp = instance.get_msrp()
+    current_price = msrp
+    price_pk = None
+
+    if price:
+        current_price = price.cost
+        price_pk = price.pk
+
+    price_data = processor.build_price(instance, msrp, current_price, DEFAULT_CURRENCY, price_pk)
+    stripe_price = processor.stripe_create_object(processor.stripe.Price, price_data)
+    instance.meta['stripe']['price_id'] = stripe_price.id
+
+    if not processor.transaction_succeded:
+        logger.error(f"stripe_create_offer_signal: {processor.transactions_info}")
+        return None
+
+    if instance.has_any_discount_or_trial():
+        coupon_data = processor.build_coupon(instance, DEFAULT_CURRENCY)
+
+        stripe_coupon = processor.stripe_create_object(processor.stripe.Coupon, coupon_data)
+        instance.meta['stripe']['coupon_id'] = stripe_coupon.id
+
+    instance.save()
+    logger.info(f"stripe_create_offer_signal instance: offer {instance.pk} successfully created on Stripe")
```

Comment:
cant you just replace this with the function we already for for creating offers?


`processor.create_offers([offer])`

Reply:
ahh